Reminder-App/Reminder-App/ReminderDetailViewController.swift

```diff
@@ -9,41 +9,139 @@ import UIKit
 
 class ReminderDetailViewController: UITableViewController {
     
+    typealias ReminderChangeAction = (Reminder) -> Void
+    
     // MARK: Private Properties
     private var reminder: Reminder?
+    private var temporaryReminder: Reminder?
     private var dataSource: UITableViewDataSource?
+    private var reminderEditAction: ReminderChangeAction?
+    private var reminderAddAction: ReminderChangeAction?
+    private var isNew = false
     
     // MARK: Public Methods
-    func configure(with reminder: Reminder) {
-        
+    func configure(with reminder: Reminder, isNew: Bool = false, addAction: ReminderChangeAction? = nil, editAction: ReminderChangeAction? = nil) {
         self.reminder = reminder
+        self.isNew = isNew
+        self.reminderAddAction = addAction
+        self.reminderEditAction = editAction
+        
+        if isViewLoaded {
+            setEditing(isNew, animated: false)
+        }
+    }
+    
+    @objc
+    func cancelButtonTrigger() {
+        if isNew {
+            dismiss(animated: true, completion: nil)
+        } else {
+            temporaryReminder = nil
+            setEditing(false, animated: true)
+        }
     }
     
     // MARK: Lifecycle Methods
     override func viewDidLoad() {
-        
         super.viewDidLoad()
-        setEditing(false, animated: false)
+        setEditing(isNew, animated: false)
         navigationItem.setRightBarButton(editButtonItem, animated: false)
         tableView.register(UITableViewCell.self, forCellReuseIdentifier: ReminderDetailEditDataSource.dateLabelCellIdentifier)
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        if let navigationController = navigationController,
+           !navigationController.isToolbarHidden {
+            navigationController.setToolbarHidden(true, animated: animated)
+        }
+    }
+    
     // MARK: Overridden Methods
     override func setEditing(_ editing: Bool, animated: Bool) {
-        
         super.setEditing(editing, animated: animated)
         
         guard let reminder = reminder else {
             return
         }
         
         if editing {
-            dataSource = ReminderDetailEditDataSource(reminder: reminder)
+            trainsitionToEditMode(reminder)
+            tableView.backgroundColor = UIColor(named: "EDIT_Background")
```

Comment:
move the **UIColor(named: "EDIT_Background")** to global colors file. and use it here like
in global colors file like
`static let editBackgroundColor =  UIColor(named: "EDIT_Background")`
and here
`tableView.backgroundColor = GlobalColors.editBackgroundColor`